pkg/api/v1alpha2/api.proto

```diff
@@ -439,9 +451,7 @@ message DeleteTrialReply {
 
 message GetTrialListRequest {
     string experiment_name = 1;
-    string start_time = 2; ///The start of the time range. RFC3339 format
-    string end_time = 3; ///The end of the time range. RFC3339 format
-    string filter = 4;
+    string filter_by_name = 2;
```

Comment:
I actually prefer "filter", because semantically it allows us to do things like:
- filter:name=foo
- filter:start_time>x
- filter:end_time<=y

We probably will only support filtering by name for now, but I can see a use case for the others. What do you think?

Reply:
Make sense. A general filter will really help. So  we need use `d.db.Query("SELECT * FROM trials WHERE experiment_name = ? AND ?, experimentName, filter)` instead, not limited in name filed filter. So when update, please comment what `filter` looks like.

Reply:
@richardsliu @hougangliu Thanks. I revert to `filter` and add comments.